src/test/java/de/tarent/resources/ParticipantFeedbackResourceTest.java

```diff
@@ -0,0 +1,161 @@
+package de.tarent.resources;
+
+import de.tarent.entities.Course;
+import de.tarent.entities.ParticipantFeedback;
+import io.quarkus.test.TestTransaction;
+import io.quarkus.test.junit.QuarkusTest;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import java.util.Set;
+
+import static io.restassured.RestAssured.given;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@QuarkusTest
+@TestMethodOrder(OrderAnnotation.class)
+class ParticipantFeedbackResourceTest {
+
+    @Test
+    public void testGetAllParticipantFeedbackForCourse() {
+        given().header(ACCEPT, APPLICATION_JSON)
+                .when().get("/courses/1/feedback")
+                .then()
+                .statusCode(200)
+                .body("participant_name", contains("Freddy Feedback"))
+                .body("likes", contains("interesting framework"))
+                .body("dislikes", contains("short live coding session"))
+                .body("recommendation", contains(true))
+                .body("any { it.any { it.key == 'id' }}", is(false))
+                .body("any { it.any { it.key == 'feedbackTime' }}", is(false));
+    }
+
+    @Test
+    public void testGetAllParticipantFeedback_UnknownCourse() {
+        given().header(ACCEPT, APPLICATION_JSON)
+                .when().get("/courses/-1/feedback")
+                .then()
+                .statusCode(404)
+                .body(is(emptyString()));
+    }
+
+    @Test
+    public void testGetAllParticipantFeedback_DeletedCourse() {
+
+        final long deletedCourseId = 3L;
+        assertTrue(Course.<Course>findById(deletedCourseId).deleted);
+
+        given().header(ACCEPT, APPLICATION_JSON)
+                .when().get("/courses/{deletedCourseId}/feedback", deletedCourseId)
+                .then()
+                .statusCode(404)
+                .body(is(emptyString()));
+    }
+
+    @Test
+    @Order(1)
```

Comment:
why is the order important here?

Reply:
Quarkus 1.9.0 has an new feature **@TestTransaction** which rolls back the transaction after a run, but I could not get it to work yet, so some tests still use the Ordering to be run before the not annotated tests. E.g. the assertions on the comment listing would not work anymore after testing the creation of new feedback